src/Microsoft.AspNet.OData.Shared/EdmEntityObject.cs

```diff
@@ -37,7 +40,61 @@ public EdmEntityObject(IEdmEntityTypeReference edmType)
         /// <param name="isNullable">true if this object can be nullable; otherwise, false.</param>
         public EdmEntityObject(IEdmEntityType edmType, bool isNullable)
             : base(edmType, isNullable)
+        {            
+            PersistentInstanceAnnotationsContainer = new ODataInstanceAnnotationContainer();
+            TransientInstanceAnnotationContainer = new ODataInstanceAnnotationContainer();            
+        }
+
+        /// <summary>
+        /// Instance Annotation container to hold Transient Annotations
+        /// </summary>
+        internal IODataInstanceAnnotationContainer TransientInstanceAnnotationContainer { get; set; }
+
+        /// <summary>
+        /// Instance Annotation container to hold Persistent Annotations
+        /// </summary>/// <inheritdoc />
+        public IODataInstanceAnnotationContainer PersistentInstanceAnnotationsContainer { get; set; }
+
+        /// <summary>
+        /// DeltaKind as Entry
+        /// </summary>
+        public virtual EdmDeltaEntityKind DeltaKind { get { return EdmDeltaEntityKind.Entry; } }
+
+        /// <summary>
+        /// To set Persistent Instance Annotation
+        /// </summary>        
+        /// <param name="value">InstanceAnnotation container value</param>
+        /// <returns>boolean representing whether the setting of instanceannotation was successful</returns>
+        public virtual bool TrySetInstanceAnnotations(IODataInstanceAnnotationContainer value)
+        {
+            PersistentInstanceAnnotationsContainer = value;
+            return true;
+        }
+
+        /// <summary>
+        /// To get Persistent Instance Annotation
+        /// </summary>        
+        /// <returns>persistence instanceannotation container </returns>
+        public virtual IODataInstanceAnnotationContainer TryGetInstanceAnnotations()
+        {
+            return PersistentInstanceAnnotationsContainer;            
+        }
+
+        /// <summary>
+        /// Method to Add Data Modification Exception
+        /// </summary>
+        public void AddDataException(DataModificationExceptionType dataModificationException)
+        {
+            TransientInstanceAnnotationContainer.AddResourceAnnotation("Core.DataModificationException", dataModificationException);           
+        }
+
+        /// <summary>
+        /// Method to Add Data Modification Exception
+        /// </summary>
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Design", "CA1024:UsePropertiesWhereAppropriate")]
+        public object GetDataException()
         {
+            return TransientInstanceAnnotationContainer.GetResourceAnnotation("Core.DataModificationException");
```

Comment:
Re: https://github.com/OData/WebApi/pull/2455/files#r616097238

Reply:
Additionally, does this method presume that `AddResourceAnnotation` was already called? Would it error if a resource annotation named "Core.DataModificationException" does not exist?

Reply:
It would return null in that case